README.md

```diff
@@ -1,28 +1,78 @@
 # github-action-detect-unused-secrets
 
-GitHub Action that automatically detects unused secrets comparing the workflow secret names with the repo secret names.
+GitHub Action that detects unused secrets.
+Compares the secret names in your repository with the secret names under `.github/workflows`
 
 ## Example usage
 
-```yaml
-name: detect-unused-secrets
+The example below runs the action every Monday at 10am.
+Generates a token with required permissions which `secrets.GITHUB_TOKEN` does not have.
+Uses checkout code to get access to the `$GITHUB_WORKSPACE`,
+And finally runs the `github-action-detect-unused-secrets` with the generated token.
 
-on: [push]
+```yml
+ name: detect-unused-secrets
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 10 * * MON'
 
 jobs:
   setup:
     runs-on: ubuntu-latest
     steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.PRIVATE_KEY }}
+      - name: Checkout Code
+      - uses: actions/checkout@v3
       - name: Detect unused secrets
-        uses: nearform/github-action-detect-unused-secrets@v1
+        uses: nearform/github-action-detect-unused-secrets@main
         with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+          github-token: ${{ steps.generate_token.outputs.token }}
+
 ```
 
 ## Usage
 
 Configure this action in your workflows providing the inputs described below.
 
+### Install a GitHub App
+
+**Required** [Create a GitHub App](https://docs.github.com/en/developers/apps/building-github-apps/creating-a-github-app) within your organisation with permission to `secrets: read-only` and [install](https://docs.github.com/en/developers/apps/managing-github-apps/installing-github-apps#installing-your-private-github-app-on-your-repository) it on the repo where you want to use this action. Save the `APP_ID` and generated `PRIVATE_KEY` to the secrets under your organisation or repo.
```

Comment:
nit: the right name for these permissions is "scope". let's keep the naming consistent

Reply:
<img width="1002" alt="Screenshot 2022-03-09 at 11 43 03" src="https://user-images.githubusercontent.com/32770804/157426014-641bffab-a8ba-49aa-a2a8-aa57c6319b5e.png">
In this case, they are names permissions, and under the permissions tab, it might be confusing to name it "scope".

Reply:
you're right. the tokens have scope, the apps have permissions. https://docs.github.com/en/rest/reference/actions#list-repository-secrets

ok let's just make sure it's consistent throughout 👍